src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -51,38 +50,43 @@ public FindCommand parse(String args) throws ParseException {
         Predicate<Person> phonePredicate = new PhoneContainsKeywordsPredicate(phoneKeywords);
         Predicate<Person> postalPredicate = new PostalContainsKeywordsPredicate(postalKeywords);
 
-        if (!nameKeywords.isEmpty() && !phoneKeywords.isEmpty() && !postalKeywords.isEmpty()) { // all 3 searches
+        System.out.println(namePredicate.getClass());
+        System.out.println(phonePredicate.getClass());
+        System.out.println(postalPredicate.getClass());
+
+        if (!nameKeywords.isEmpty() && !phoneKeywords.isEmpty() && !postalKeywords.isEmpty()) {
```

Comment:
This section appears to have alot of branching logic with repeated checks. Could this be possibly refactored?